python/lsst/analysis/tools/actions/vector/calcRhoStatistics.py

```diff
@@ -0,0 +1,649 @@
+            ),
+            "Arc": "the true great circle distance for spherical coordinates.",
+            "Periodic": "Like ``Euclidean``, but with periodic boundaries.",
+        },
+    )
+
+    bin_type = ChoiceField[str](
+        doc="What type of binning should be used?",
+        default="Log",
+        optional=True,
+        allowed={
+            "Log": (
+                "Logarithmic binning in the distance. The bin steps will "
+                "be uniform in log(r) from log(min_sep) .. log(max_sep)."
+            ),
+            "Linear": (
+                "Linear binning in the distance. The bin steps will be "
+                "uniform in r from min_sep .. max_sep."
+            ),
+            "TwoD": (
+                "2-dimensional binning from x = (-max_sep .. max_sep) "
+                "and y = (-max_sep .. max_sep). The bin steps will be "
+                "uniform in both x and y. (i.e. linear in x,y)"
+            ),
+        },
+    )
+
+    min_rpar = Field[float](
+        doc=(
+            "The minimum difference in Rparallel to allow for pairs being "
+            "included in the correlation function. "
+        ),
+        default=None,
+        optional=True,
+    )
+
+    max_rpar = Field[float](
+        doc=(
+            "The maximum difference in Rparallel to allow for pairs being "
+            "included in the correlation function. "
+        ),
+        default=None,
+        optional=True,
+    )
+
+    period = Field[float](
+        doc="For the 'Periodic' metric, the period to use in all directions.",
+        default=None,
+        optional=True,
+    )
+
+    xperiod = Field[float](
+        doc="For the 'Periodic' metric, the period to use in the x direction.",
+        default=None,
+        optional=True,
+    )
+
+    yperiod = Field[float](
+        doc="For the 'Periodic' metric, the period to use in the y direction.",
+        default=None,
+        optional=True,
+    )
+
+    zperiod = Field[float](
+        doc="For the 'Periodic' metric, the period to use in the z direction.",
+        default=None,
+        optional=True,
+    )
+
+    var_method = ChoiceField[str](
+        doc="Which method to use for estimating the variance",
+        default="shot",
+        optional=True,
+        allowed={
+            method: method
+            for method in [
+                "shot",
+                "jackknife",
+                "sample",
+                "bootstrap",
+                "marked_bootstrap",
+            ]
+        },
+    )
+
+    num_bootstrap = Field[int](
+        doc=("How many bootstrap samples to use for the 'bootstrap' and " "'marked_bootstrap' var methods."),
+        default=500,
+        optional=True,
+    )
+
+    num_threads = Field[int](
+        doc=(
+            "How many OpenMP threads to use during the calculation. "
+            "If set to None, use all the available CPU cores."
+        ),
+        default=None,
+        optional=True,
+    )
```

Comment:
I'm not sure if you want num_threads to be settable either.  Probably that should be set based on something in the DM configuration elsewhere, rather than let the user set it explicitly here, but I don't know enough to say what to recommend on that one.

Reply:
Removed a bunch of fields incl. `num_threads`, because it is easier to add fields and even backport them, than to remove fields.